backstory/src/test/java/com/google/sps/vision/VisionManagerImplTest.java

```diff
@@ -0,0 +1,159 @@
+  /*
+   * public VisionManagerImpl(byte[] rawImageData);
+   *  Partition on rawImageData: empty, null, non-empty image data.
+   * 
+   * public String getLabelsAsJson();
+   *  Unnecesary to test as it is a wrapper for a Gson call.
+   *
+   * public List<String> getLabelDescriptions();
+   *  Partition on this: labelAnnotations is empty, or non-empty list of EntityAnnotations.
+   */
+
+  private byte[] nullRawImageData;
+  private byte[] emptyRawImageData;
+  private byte[] rawImageData;
+  private List<EntityAnnotation> emptyLabelAnnotations;
+  private List<EntityAnnotation> labelAnnotations;
+
+  @Before
+  public void setUp() throws IOException {
+    nullRawImageData = null;
+    emptyRawImageData = new byte[0];
+    rawImageData = getBytesFromImageReference(
+      "src/test/java/com/google/sps/vision/data/dogRunningOnBeach.jpg", "jpg");
+    emptyLabelAnnotations = new ArrayList<>();
+
+    labelAnnotations = new ArrayList<>();
+
+    EntityAnnotation entityAnnotationMock = mock(EntityAnnotation.class);
+    // mock the behavior of stock service to return the value of various stocks
+    when(entityAnnotationMock.getDescription()).thenReturn("DescriptionOne");
+
+    labelAnnotations.add(entityAnnotationMock);
+    labelAnnotations.add(entityAnnotationMock);
+  }
+
+  /**
+   * Returns an image as a byte array from the local reference of the image.
+   *
+   * @param reference the local reference of the image.
+   * @param fileType the file extension of the image.
+   * @return the byte array representation of the image.
+   */
+  private static byte[] getBytesFromImageReference(String reference, String fileType) throws IOException {
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+    try {
+      BufferedImage bImage = ImageIO.read(new File(reference));
+      ImageIO.write(bImage, fileType, bos);
+    } catch (IOException exception){
+      throw exception;
+    }
+
+    return bos.toByteArray();
+  }
+
+  /**
+   * Tests paritions on public VisionManagerImpl(byte[] rawImageData);
+   *  rawImageData is null.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorNullImageData() throws IOException {
+    VisionManager visionManagerNullImage = new VisionManagerImpl(nullRawImageData);
+  }
+
+  /**
+   * Tests paritions on VisionManagerImpl(byte[] rawImageData);
+   *  rawImageData is empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorEmptyImageData() throws IOException {
+    VisionManager visionManagerNullImage = new VisionManagerImpl(emptyRawImageData);
+  }
+
+  /**
+   * Tests paritions on VisionManagerImpl(byte[] rawImageData);
+   *  rawImageData is non-empty image data.
+  */
+  @Test
+  public void constructorRawImageData() {
+    VisionManagerImpl actual = new VisionManagerImpl(rawImageData, labelAnnotations);
```

Comment:
Does this make a network call? If not, could you point me to where the network call is stubbed out?